sys/include/net/netopt.h

```diff
@@ -574,6 +577,84 @@ typedef enum {
      */
     NETOPT_PHY_BUSY,
 
+    /**
+     * @brief   (uint8_t*) LoRaWAN device EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_DEVEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_APPEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPKEY,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN network session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_NWKSKEY,
+    /**
+     * @brief   (uint8_t*) LoRaWAN application session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPSKEY,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN join state
+     */
+    NETOPT_LORAWAN_JOIN_STATE,
+
+     /**
+     * @brief   (uint8_t) LoRaWAN device class (A, B, C)
+     * - LoRaWAN: @ref loramac_class_t
+     */
+    NETOPT_LORAWAN_DEVICE_CLASS,
+
+    /**
+     * @brief   (uint8_t) LoRaWAN datarate
+     * - LoRaWAN: @ref loramac_dr_idx_t
+     */
+    NETOPT_LORAWAN_DATARATE,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN adaptive datarate
```

Comment:
I think I would short it to "NETOPT_LORAWAN_ADR"

Reply:
I think this is bound to be confused with `ADDR`, but I'm not blocking because of this.

Reply:
the thing is the netopt for that is NETOPT_ADDRESS, and ADR is a well known feature in LoRaWAN. It's fine for me

Reply:
@miri64 are you OK with this?

Reply:
I'm not questioning that... Just noticed it when I read the thread, that my brain automatically jumped to `ADDR`. But as I said: this is not the reason for me blocking ;-)

Reply:
Sorry, never commitet my answer to https://github.com/RIOT-OS/RIOT/pull/9022/files#r297086059 and did not see your question, so now my answer looks weird ^^